_posts/2019-10-27-on-code-reviews.md

```diff
@@ -0,0 +1,102 @@
+- **I assume best intent.** I've written questionable code before, or have
+overseen mistakes. That happens. Always be kind and approach a review
+positively. 
+- **I read the pull request's description _and_ ticket.** Yes, also the ticket.
+This makes the time of everyone involved a lot more valuable. Seldom does this
+not help me getting a better understanding of the code. And understanding is
+key. If something in the description or ticket is unclear to me, I try to reach
+out the author before continuing a review.
+- **I provide context.** For instance, if I come across something that might
+better be represented by a function rather than a computed property, I explain
+why I would suggest that in the given case.
+- **I provide references.** Whether it's a blog post, piece of code, or
+documentation I reference, I link it. This allows the author to dig deeper into
+the topic if they want, and may be helpful for those who read the comment(s) in
+the future.
+- **I take the time to try and understand the code.** When I feel like I'm
+having a hard time grasping some of the decisions that have been made, or the
+code itself, I often feel asking about it in writing is difficult. If this
+happens, I either take a step away from the computer and try again later, or ask
+to go over it in person (face-to-face or in a video call, _not_ a phone call.
+Being able to see each other aids understanding).<br />
+An exception to this would be a pull request I feel is introducing too many
+concepts at once, in which case I'll either ask if we could break up the pull
+request in more bite-sized pieces, offering to help do so in person if need be.
+- **I ask questions.** I would call myself a curious person. If I see something
+I don't know about, or want to know more about (e.g. a piece of code, business
+logic), I ask. If the review already has many comments, I might do this
+outside the review itself.<br />
+Also, if I see code that appears to be copied, where I feel improvements can be
+made, I ask if this is something that has been considered/if this is something
+for us to consider in a follow-up pull request.
+- **I re-read a review before I submit it.** I don't always do this, but I
+know it helps. I've seen the author ask about comments I wrote, not (fully)
+understanding what I meant. Though communication is hard, I feel this has
+been happening less over time.
+- **I feel responsible for the code I review,** meaning I use "we", not
+"you". "We could consider adding a comment explaining why we need to ignore X",
+not "you".
+- **I review my own code.** Especially in larger pull requests, I look through
+the code before assigning a reviewer, asking open questions to the reviewer,
+making notes and/or making some changes. Seeing the "bigger" picture often helps
+me catch issues/improvements to be made early, hopefully making the code review
+more enjoyable in the process.
+- **I don't use words like "easy", "simply", or similar.** The fact something
```

Comment:
IMO similar is fine to use and very different from the others. "Would it be possible to abstract this similarly to how we did that other thing?" or "This looks like it solves a problem similar to that other extension we have in the codebase. Would it make sense to unify?"

Reply:
Ah; I was only pointing out "easy" and "simply". "similar" is meant to indicate that list isn't exhaustive. I'll reword, as it seems confusing. :)